From 30064fc0ddf28a809d134d6d6255dda687e3df50 Mon Sep 17 00:00:00 2001
From: Tirumalesh Chalamarla <tchalamarla@caviumnetworks.com>
Date: Tue, 16 Oct 2018 12:51:00 +0300
Subject: [PATCH 0271/1051] octeontx-sso: fix get_vf

get_vf should return proper VF not the subdomain_id 0.

Signed-off-by: Santosh Shukla <santosh.shukla@caviumnetworks.com>
Signed-off-by: Tirumalesh Chalamarla <tchalamarla@caviumnetworks.com>
Signed-off-by: Yury Norov <ynorov@marvell.com>
Signed-off-by: Yury Norov <ynorov@caviumnetworks.com>
[Kevin: The original patch got from Marvell sdk10.0_19.06]
Signed-off-by: Kevin Hao <kexin.hao@windriver.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/ssopf_main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/ssopf_main.c b/drivers/net/ethernet/cavium/octeontx-83xx/ssopf_main.c
index 7f3f40ea09be..23fb7b51cd04 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/ssopf_main.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/ssopf_main.c
@@ -76,7 +76,7 @@ static struct ssopf_vf *get_vf(struct ssopf *sso, u16 domain_id,
 	for (i = 0; i < sso->total_vfs; i++) {
 		if (sso->vf[i].domain.in_use &&
 		    sso->vf[i].domain.domain_id == domain_id &&
-		    sso->vf[i].domain.subdomain_id == 0) {
+		    sso->vf[i].domain.subdomain_id == subdomain_id) {
 			if (vf_idx)
 				*vf_idx = i;
 			return &sso->vf[i];
-- 
2.17.1

